source/common/upstream/upstream_impl.cc

```diff
@@ -110,14 +111,25 @@ ClusterInfoImpl::ClusterInfoImpl(const envoy::api::v2::Cluster& config,
       maintenance_mode_runtime_key_(fmt::format("upstream.maintenance_mode.{}", name_)),
       source_address_(getSourceAddress(config, source_address)),
       lb_ring_hash_config_(envoy::api::v2::Cluster::RingHashLbConfig(config.ring_hash_lb_config())),
-      added_via_api_(added_via_api),
+      ssl_context_manager_(ssl_context_manager), added_via_api_(added_via_api),
       lb_subset_(LoadBalancerSubsetInfoImpl(config.lb_subset_config())) {
-  ssl_ctx_ = nullptr;
-  if (config.has_tls_context()) {
-    Ssl::ClientContextConfigImpl context_config(config.tls_context());
-    ssl_ctx_ = ssl_context_manager.createSslClientContext(*stats_scope_, context_config);
+
+  auto transport_socket = config.transport_socket();
```

Comment:
I think I'm being a little dense here, but doesn't `transport_socket()` return a const object as per the proto docs? How do the calls below work?

Reply:
I guess maybe it copies by value? That must be it.

Reply:
Yes it copies by value since the codes below have to modify it.